drivers/ThinQ_Washer.groovy

```diff
@@ -42,74 +57,90 @@ def initialize() {
+                            mqtt.clientId,
+                            null,
+                            null,
+                            tlsVersion: "1.2",
+                            privateKey: mqtt.privateKey,
+                            caCertificate: mqtt.caCertificate,
+                            clientCertificate: mqtt.certificate)
+    pauseExecution(3000)
+    for (sub in mqtt.subscriptions) {
+        interfaces.mqtt.subscribe(sub, 0)
+    }
 		return true
 	}
 	catch (e)
 	{
-		log.warn "Lost connection to MQTT, retrying in 15 seconds ${e}"
+    logger("warn", "Lost connection to MQTT, retrying in 15 seconds ${e}")
 		runIn(15, "mqttConnectUntilSuccessful")
 		return false
 	}
 }
 
 def parse(message) {
+    logger("debug", "parse(${message})")
     def topic = interfaces.mqtt.parseMessage(message)
-    def payload = new JsonSlurper().parseText(topic.payload) 
+    def payload = new JsonSlurper().parseText(topic.payload)
 
     parent.processMqttMessage(this, payload)
 }
 
 def mqttClientStatus(String message) {
-    log.debug "Status: " + message
+    logger("debug", "mqttClientStatus(${message})")
 
     if (message.startsWith("Error:")) {
-        log.error "MQTT Error: ${message}"
+        logger("error", "MQTT Error: ${message}")
+
         try {
             interfaces.mqtt.disconnect() // Guarantee we're disconnected
         }
         catch (e) {
         }
-		mqttConnectUntilSuccessful()
+        mqttConnectUntilSuccessful()
     }
 }
 
 def processStateData(data) {
     def runTime = 0
     def remainingTime = 0
     def currentState = data["State"]
-    def error 
+    def error
     def soilLevel = data["Soil"] ?: ""
     def spinSpeed = data["SpinSpeed"] ?: ""
     def waterTemp = data["WaterTemp"] ?: ""
 
-    remainingTime += (data["Remain_Time_H"]*60*60)
-    remainingTime += (data["Remain_Time_M"]*60)
+    if (data?.containsKey('Remain_Time_H') ) {
+      remainingTime += (data["Remain_Time_H"]*60*60)
+      remainingTime += (data["Remain_Time_M"]*60)
+    }
 
-    runTime += (data["Initial_Time_H"]*60*60)
-    runTime += (data["Initial_Time_M"]*60)
+    if (data?.containsKey('Initial_Time_H') ) {
+      runTime += (data["Initial_Time_H"]*60*60)
+      runTime += (data["Initial_Time_M"]*60)
+    }
+
+    if (data?.containsKey('type') && data.type == "monitoring") {
+        sendEvent(name: "switch", value: data.state.reported.online =~ /true/ ? 'on' : 'off')
+    }
 
     sendEvent(name: "runTime", value: runTime)
     sendEvent(name: "runTimeDisplay", value: "${data["Remain_Time_H"]}:${data["Remain_Time_M"]}")
     sendEvent(name: "remainingTime", value: remainingTime)
     sendEvent(name: "remainingTimeDisplay", value: "${data["Initial_Time_H"]}:${data["Initial_Time_M"]}")
     if (currentState != null)
         sendEvent(name: "currentState", value: parent.cleanEnumValue(currentState, "@WM_STATE_"))
-    sendEvent(name: "error", value: data["Error"].toLowerCase())
+
+    if (data?.containsKey('Error') ) {
```

Comment:
I'm curious about this if statement. My thought was, if there is no error reported *now* we might want to clear out the old error. I don't know exactly what kind of errors we'll get but my thinking is, say you get an error like "door stuck open." When you close the door, we probably get null for Error so we want to clear the attribute.